### PR TITLE
Added test project

### PR DESCRIPTION
**Pull request by MissedTheMark**
Added support for nested groups, as well as some code improvements (added TinyTypes for group DN)

### CODE REVIEW

File: source/Server/Ldap/NestedGroupFinder.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+        public NestedGroupFinder(IGroupParentFinder parentFinder)
+        {
+            this.parentFinder = parentFinder;
+        }
+
+        public IEnumerable<GroupDistinguishedName> FindAllParentGroups(LdapContext context, int searchDepth, IEnumerable<GroupDistinguishedName> names)
+        {
+            var groups = new HashSet<GroupDistinguishedName>(names.Distinct());
+            var nextGroups = groups.ToList();
```
Comment (tothegills) [inline source/Server/Ldap/NestedGroupFinder.cs]: Curious why `ToList` here?

File: source/Ldap.Tests/NestedGroupFinderTests.cs [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+using NSubstitute;
+using Octopus.Server.Extensibility.Authentication.Ldap;
+using Octopus.Server.Extensibility.Authentication.Ldap.Model;
+using Xunit;
+
+namespace Ldap.Tests
+{
+    public class NestedGroupFinderTests
+    {
+        [Theory]
```
Comment (tothegills) [inline source/Ldap.Tests/NestedGroupFinderTests.cs:13]: I think the other extensions are Nunit, what you made you choose Xunit?

**Review by tothegills — APPROVED** [review]
Do the tests run as part of the build?

I'm assuming the correctness of the tests. Looks good to me, a couple of inline questions to consider.

File: source/Server/Ldap/NestedGroupFinder.cs (hunk 1 above)
Comment (MissedTheMark) [reply source/Server/Ldap/NestedGroupFinder.cs]: Good pick up - I have no idea 🤷‍♂️ 
Fixed it now.

**Comment by MissedTheMark** [comment]
> Do the tests run as part of the build?

Sure do!

File: source/Ldap.Tests/NestedGroupFinderTests.cs (hunk 2 above)
Comment (MissedTheMark) [reply source/Ldap.Tests/NestedGroupFinderTests.cs:13]: AFAIK, we're using Xunit going forward.

**Pull request merged**